File: Hackathon.UI/src/app/models/Event/EventStatus.ts
export enum EventStatus {
  Draft = 0,
  OnModeration = 3,
  Published = 1,
  Started = 2,
  Finished = 8,
}

export class EventStatusTranslator {
  public static GetName = (e:EventStatus) : string => EventStatus[e].toLowerCase();
  public static Translate = (e: EventStatus) : string => {
    switch (e) {
      case EventStatus.Draft: return 'Черновик'
      case EventStatus.OnModeration: return 'На модерации'
      case EventStatus.Published: return 'Опубликовано'
      case EventStatus.Started: return 'Начато'
      case EventStatus.Finished: return 'Завершено'

      default: return EventStatus[e];
    }
  }
}

File: src/Hackathon.Common/Models/Event/EventStatus.cs
namespace Hackathon.Common.Models.Event;

/// <summary>
/// Статус мероприятия
/// </summary>
public enum EventStatus
{
    /// <summary>
    /// Черновик
    /// </summary>
    Draft = default,

    /// <summary>
    /// На модерации
    /// </summary>
    OnModeration = 3,

    /// <summary>
    /// Опубликовано
    /// </summary>
    /// <remarks>
    /// Нельзя вносить изменения в общие параметры
    /// </remarks>
    Published = 1,

    /// <summary>
    /// Событие начато
    /// </summary>
    /// <remarks>Регистрация закрыта</remarks>
    Started = 2,

    /// <summary>
    /// Событие завершено
    /// </summary>
    /// <remarks>Все доступные ранее операции запрещены</remarks>
    Finished = 8
}
